smp/enumeration_management.py

```diff
@@ -0,0 +1,123 @@
+"""The Simple Management Protocol (SMP) Enumeration Management group."""
+
+from __future__ import annotations
+
+from enum import IntEnum, unique
+from typing import Tuple
+
+from pydantic import BaseModel, ConfigDict
+
+import smp.error as smperr
+import smp.header as smphdr
+import smp.message as smpmsg
+
+
+class GroupCountRequest(smpmsg.ReadRequest):
+    """Read the number of SMP server groups."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_COUNT
+
+
+class GroupCountResponse(smpmsg.ReadResponse):
+    """SMP group count response."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_COUNT
+
+    count: int
+
+
+class ListOfGroupsRequest(smpmsg.ReadRequest):
+    """List the available SMP groups."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.LIST_OF_GROUPS
+
+
+class ListOfGroupsResponse(smpmsg.ReadResponse):
+    """SMP group list response."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.LIST_OF_GROUPS
+
+    groups: Tuple[int, ...]
+
+
+class GroupIdRequest(smpmsg.ReadRequest):
+    """List a SMP group by index."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_ID
+
+    index: int | None = None
+
+
+class GroupIdResponse(smpmsg.ReadResponse):
+    """SMP group at index response."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_ID
+
+    group: int
```

Comment:
Try:

```python
group: GroupId | UserGroupId | int = Field(union_mode='left_to_right')
```

Might be easier to define the "Group" type for this file.

Reply:
https://github.com/JPHutchins/smp/blob/43ee844ddaf5bcdb8dda2b81c52a95438833aa69/smp/message.py#L31C25-L31C89